### PR TITLE
Doc updates and fixes

### PR DESCRIPTION
**Pull request by dzimine**

### CODE REVIEW

**Comment by manasdk** [comment]
+1

FYI : Fixed docs build.

File: docs/source/install/on_complete.rst [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-config file <cli-configuration>` or provided as an environment variable.
-
 Check that |st2| installation works OK: ::
 
     st2ctl status
 
     st2 --version
     st2 --help
+
+    # If AUTH enabled: authenticate and export the token env variable
```
Comment (Kami) [inline docs/source/install/on_complete.rst]: Latest version (0.11.0) now also supports CLI config so we should tell users to store the credentials in the config instead (docs/source/cli.rst) if they don't want to re-authenticate on every request.

File: docs/source/start.rst [hunk 2]
```diff
@@ -35,6 +37,17 @@ Check the :doc:`CLI and Python Client </reference/cli>` reference for details.
 From v0.8, |st2| ships with WebUI. With the default "all-in-one" installation, you can access it at
 http://hostname:8080/.
 
+Authenticate
+----------------
+If :doc:`authentication </authentication>` enabled, obtain authentication token with ``st2 auth <username>``,
+and supply it with each command using ``--token`` parameter. For convinience,
```
Comment (Kami) [inline docs/source/start.rst]: typo - convenience

File: docs/source/install/index.rst [hunk 3]
```diff
@@ -1,10 +1,10 @@
 Installation
 ============
-There are a few ways to simply download and install StackStorm to get going.  
+There are a few ways to install StackStorm to get going.  The easiest is all-in-one install
+on Ubuntu/Debian or RedHat/Fedora which we explain next. Or skip below if you want to learn about
+:doc:`docker` or :doc:`vagrant`, or other installations approaches.
```
Comment (Kami) [inline docs/source/install/index.rst]: approaches / methods?

**Comment by Kami** [comment]
Closing since it looks like #1613 supersedes this PR.

My comments are still relevant though :)

**Comment by manasdk** [comment]
@kami This PR is for master and 1613 is for 0.11.2 - we need both. Reopening.

**Comment by Kami** [comment]
Ah, sorry, I missed that.

Usually it's easier if we just create one PR against version branch and then merge version branch into master.

**Comment by dzimine** [comment]
> Usually it's easier if we just create one PR against version branch and then merge version branch into master.
Ok, will try next time.

**Comment by dzimine** [comment]
@Kami did the fixes on both branches, except one in `on_complete.rst` as I just want user to quickly test that stuff works there, educate later.

**Pull request merged**